### ISSUE TITLE
reduce/reduce conflicts report shows same reduction twice

### ISSUE THREAD

**Issue by osa1** [lalrpop/lalrpop#685]
Grammar:

```rust
grammar;

extern {
    enum Token {
        "(" => Token::LParen, // 0
        ")" => Token::RParen,
        "bool" => Token::Bool(<bool>),
        "not" => Token::Not,
        "if" => Token::If,
        "then" => Token::Then, // 5
        "else" => Token::Else,
        "let" => Token::Let,
        "rec" => Token::Rec,
        "in" => Token::In,
        "-" => Token::Minus, // 10
        "-." => Token::MinusDot,
        "+" => Token::Plus,
        "+." => Token::PlusDot,
        "*." => Token::AstDot,
        "/." => Token::SlashDot, // 15
        "=" => Token::Equal,
        "<>" => Token::LessGreater,
        "<=" => Token::LessEqual,
        "<-" => Token::LessMinus,
        "<" => Token::Less, // 20
        ">=" => Token::GreaterEqual,
        ">" => Token::Greater,
        "." => Token::Dot,
        "," => Token::Comma,
        ";" => Token::Semicolon, // 25
        "_" => Token::Underscore,
        "Array.create" => Token::ArrayCreate,
        "id" => Token::Id(<String>), // 28
        "int" => Token::Int(<i64>),
        "float" => Token::Float(<f64>), // 30
    }
}

pub Expr: ParsedExpr = {
    #[precedence(level = "1")]
    "(" ")" =>
        ParsedExpr::Unit,

    "(" <expr:Expr> ")" =>
        expr,

    <b:"bool"> =>
        ParsedExpr::Bool(b),

    <i:"int"> =>
        ParsedExpr::Int(i),

    <f:"float"> =>
        ParsedExpr::Float(f),

    <var:"id"> =>
        ParsedExpr::Var(var),

    "let" <bndr:Binder> "=" <rhs:Expr> "in" <body:Expr> =>
        ParsedExpr::Let { bndr, rhs: Box::new(rhs), body: Box::new(body) },

    "let" "rec" <bndr:Binder> <args:Binders1> "=" <rhs:Expr> "in" <body:Expr> =>
        ParsedExpr::LetRec { bndr, args, rhs: Box::new(rhs), body: Box::new(body) },

    "let" <bndr:Binder> "=" <rhs:Expr> "in" <body:Expr> =>
        ParsedExpr::Let { bndr, rhs: Box::new(rhs), body: Box::new(body) },
};

Binder: String = {
    "_" =>
        "_".to_owned(),

    <id:"id"> =>
        id,
};

// One or more identifiers
Binders1: Vec<String> = {
    <bndr:Binder> <mut bndrs:Binders0_Rev> => {
        bndrs.push(bndr);
        bndrs.reverse();
        bndrs
    }
};

// Zero or more identifiers
Binders0_Rev: Vec<String> = {
    => vec![],

    // TODO: I don't understand why I have to inline Binder here.
    <bndr:"id"> <mut bndrs:Binders0_Rev> => {
        bndrs.push(bndr);
        bndrs
    }

    "_" <mut bndrs:Binders0_Rev> => {
        bndrs.push("_".to_owned());
        bndrs
    }
};
```

Error:

```
src/parser.lalrpop:59:5: 60:74: Ambiguous grammar detected

  The following symbols can be reduced in two ways:
    "(" "let" Binder "=" Expr "in" Expr ")"

  They could be reduced like so:
    "(" "let" Binder "=" Expr "in" Expr ")"
    │   └─Expr────────────────────────┘   │
    └─Expr────────────────────────────────┘

  Alternatively, they could be reduced like so:
    "(" "let" Binder "=" Expr "in" Expr ")"
    │   └─Expr────────────────────────┘   │
    └─Expr────────────────────────────────┘

  LALRPOP does not yet support ambiguous grammars. See the LALRPOP manual for advice on making your grammar unambiguous.
```

The example in "alternatively" is the same as the example before.

**Comment by andersk** [comment]
You wrote two copies of this rule:

```rust
pub Expr: ParsedExpr = {
    …
    "let" <bndr:Binder> "=" <rhs:Expr> "in" <body:Expr> =>
        ParsedExpr::Let { bndr, rhs: Box::new(rhs), body: Box::new(body) },
    …
    "let" <bndr:Binder> "=" <rhs:Expr> "in" <body:Expr> =>
        ParsedExpr::Let { bndr, rhs: Box::new(rhs), body: Box::new(body) },
};
```

**Comment by trueblablak** [comment]
This is still an issue with error messages, I think it should be reopened

**Comment by dburgener** [comment]
This was addressed in #927 and should be fixed in the forthcoming 0.22 release.
